dbms/src/Functions/FunctionsHashing.h

```diff
@@ -956,6 +956,136 @@ class FunctionURLHash : public IFunction
     }
 };
 
+struct JavaHashImpl
+{
+    static Int32 apply(const char * data, const size_t size)
+    {
+        Int32 h = 0;
+        for (int i = 0; i < (int)size; ++i)
```

Comment:
Why we use signed index?
And this code lack support for > 2GB strings.

Reply:
In java, they use Int32 type, so, the hash value can less than 0, just for compatible

Reply:
I mean type of loop variable.